tests/ha/single_pod_restart/test_data_server_pod_restart.py

```diff
@@ -96,23 +97,63 @@ def setup_method(self):
+        self.deployment_backup = None
+        self.num_replica = 1
         self.s3acc_name = f"ha_s3acc_{int(perf_counter_ns())}"
         self.s3acc_email = f"{self.s3acc_name}@seagate.com"
         LOGGER.info("Precondition: Verify cluster is up and running and all pods are online.")
         resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
         assert_utils.assert_true(resp[0], resp[1])
         LOGGER.info("Precondition: Verified cluster is up and running and all pods are online.")
+        convert = lambda text: int(text) if text.isdigit() else text
+        alphanum_key = lambda key: [convert(c) for c in re.split('([0-9]+)', key)]
+        LOGGER.info("Get %s and %s pods to be deleted", const.POD_NAME_PREFIX,
+                    const.SERVER_POD_NAME_PREFIX)
+        self.pod_dict = dict()
+        for prefix in [const.POD_NAME_PREFIX, const.SERVER_POD_NAME_PREFIX]:
+            self.pod_list = list()
+            sts_dict = self.node_master_list[0].get_sts_pods(pod_prefix=prefix)
+            sts_list = list(sts_dict.keys())
+            LOGGER.debug("%s Statefulset: %s", prefix, sts_list)
+            sts = self.system_random.sample(sts_list, 1)[0]
+            sts_dict_val = sorted(sts_dict.get(sts), key=alphanum_key)
+            self.delete_pod = sts_dict_val[-1]
+            LOGGER.info("Pod to be deleted is %s", self.delete_pod)
+            self.set_type, self.set_name = self.node_master_list[0].get_set_type_name(
+                pod_name=self.delete_pod)
+            self.pod_list.append(self.delete_pod)
+            self.pod_list.append(self.set_name)
+            resp = self.node_master_list[0].get_num_replicas(self.set_type, self.set_name)
+            assert_utils.assert_true(resp[0], resp)
+            self.num_replica = int((resp[1]))
+            self.pod_list.append(self.num_replica)
+            self.pod_dict[prefix] = self.pod_list
         LOGGER.info("COMPLETED: Setup operations. ")
 
     def teardown_method(self):
         """
         This function will be invoked after each test function in the module.
         """
+        LOGGER.info("STARTED: Teardown Operations.")
+        if self.restore_pod:
+            for pod in self.pod_dict:
+                self.restore_method = self.pod_dict.get(pod)[-1]
+                resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                               restore_method=self.restore_method,
+                                               restore_params={
+                                                   "deployment_name": self.pod_dict.get(pod)[-2],
+                                                   "deployment_backup": self.deployment_backup,
+                                                   "num_replica": self.pod_dict.get(pod)[2],
+                                                   "set_name": self.pod_dict.get(pod)[1]})
```

Comment:
self.pod_dict has prefixes as keys right?
Here you are using pod as key, this won't work. Have you tried running this?